community/primitive-collections/src/main/java/org/neo4j/concurrent/CompletableFuture.java

```diff
@@ -143,4 +143,8 @@ public T get( long timeout, TimeUnit unit ) throws InterruptedException, Executi
         }
         return state.get().get();
     }
+
+    public abstract static class AutoCloseable<T> extends CompletableFuture<T> implements AutoCloseableFuture<T>
```

Comment:
Do not think we should extend our own `CompletableFuture` but instead we should switch to `java.util.concurrent.CompletableFuture` everywhere.